### DISCUSSION TITLE
SystaPi on Raspberry Pi 3 Model B in Home Network without Wifi

### DISCUSSION THREAD

**Discussion by dehessijames** [beep-projects/SystaPi#15] [Q&A]
Hello,
first of all thank you very much for this great project. 

I tried to install it on a Raspberry Pi 3 Model B Rev 1.2, but I cannot connect the Raspberry directly to the SystaComfort II, because this connection is necessary for the pellets boiler. However, the pellet boiler and the SystaComfort II are connected to my home network via cable.

Therefore I installed the Project without Wifi-Config, DNSMASQ and DHCP. 

The /home/beep/SystaRESTServer/bin/SystaREST.properties I have adapted:
RESTAPI_INTERFACE=eth0
#RESTAPI_PORT should be greater than 1024. Only root can open lower port numbers
RESTAPI_PORT=1337
PARADIGMA_INTERFACE=eth0

I have configured the SystaComfort II according to the following instructions https://github.com/xgcssch/SystaComfortPrometheusExporter#configure-systacomfort-controller

Now te call of http://systapi:1337/SystaREST/findsystacomfort returns the following result

<html><body>

SystaWebIP | "192.168.0.xxx"
-- | --
SystaWebPort | 22460
DeviceTouchBcastIP | "192.168.0.xxx"
DeviceTouchBcastPort | 8001
deviceTouchInfoString | "SC2 1 192.168.0.xxx 255.255.255.0 192.168.0.xxx SystaComfort-II\u00000 080982xxxx 0 V1.02.1 V1.00 2CBE9701xxxx"
unitIP | "192.168.0.5"
unitName | "SystaComfort-II"
unitId | "080982xxxx"
unitApp | 8
unitPlatform | 9
unitVersion | "1.3.2"
unitMajor | 130
unitMinor | 2
unitBaseVersion | "V1.02.1"
unitMac | "2CBE9701xxxx"
STouchAppSupported | false
DeviceTouchPort | -1
DeviceTouchPassword | "null"

</body>
</html>

The call of http://systapi:1337/SystaREST/servicestatus returns the following result
<html><body>

timeStampString | "2022-11-07T22:22:35.940314+01:00"
-- | --
connected | false
running | true
lastDataReceivedAt | "never"
packetsReceived | 0
paradigmaListenerIP | "192.168.0.xxx"
paradigmaListenerPort | 22460
paradigmaIP | ""
paradigmaPort | 0
loggingData | false
logFileSize | 60
logFilePrefix | "DataLogger"
logFileDelimiter | ";"
logFileRootPath | "/home/beep/logs"
logFilesWritten | 0
logBufferedEntries | 0
commitDate | "2022-04-20T20:33:54+00:00"

</body>
</html>

Unfortunately, "lastDataReceivedAt"-Value is permanently set to "never" and so far I have not been able to find a way to test whether the SystaComfort II really does not send any data.

Is my idea even possible or do I have a thinking error? 

I would be very glad about help!

**Comment by beep-projects** [comment]
Hi @dehessijames,
thanks for giving SystaPi a try!
Your setup should be possible, but it requires the right configuration of the network.
There are two fundamental things to check on the network side before digging deeper.
1. Did you configure your network to route the traffic intended for _paradigma.remoteportal.de_ to the IP configured for eth0 on your Raspberry Pi?
I assume yes, because it is the step before the linked config https://github.com/xgcssch/SystaComfortPrometheusExporter#configure-your-dns
2. Do you see packets from the SystaComfort arriving at eth0? e.g. check with `sudo tcpdump -A -i eth0 `
you should see something like
```
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on eth0, link-type EN10MB (Ethernet), snapshot length 262144 bytes
09:14:36.274393 IP SystaComfort-II.8002 > 192.168.0.xxx.22460: UDP, length 1048
...
```

**Reply by dehessijames** [reply]
Hi @beep-projects ,
Thank you for the quick reply!

To 1.:
Sorry I forgot to mention that. I'm already running another PI in my network with nextcloudpi an DNSMASQ on it. There I have configured paradigma.remoteportal.de to the ip of the systapi. DNS resolution of paradigma.remoteportal.de returns my local ip 192.168.0.xxx of the systapi (tested on several devices in my network).

To 2.:
`sudo tcpdump -A -i eth0 ` returns a lot of output, because I'm using ssh but I've allready tried `tcpdump udp`. There I can see no connections of the SystaComfort II. Maybe the SystaComfort II really does not send any data and  I've to poweroff and restart it, or something.

**Comment by beep-projects** [comment]
If your SystaComfort is prepared for SystaWeb, you should receive a group of 3-4 UDP packets every minute.
Be aware, that you have to enable the communication with SystaWeb on the SystaComfort AND on the attached control unit.
There is a discussion in German on how to do this 
https://github.com/xgcssch/SystaComfortPrometheusExporter/issues/6#issuecomment-988774072

**Comment by dehessijames** [comment]
Yes I've done that both. And on the PI with DNSMASQ I can see the DNS Request:
Nov  8 13:15:45 nextcloudpi dnsmasq[11696]: query[A] paradigma.remoteportal.de from 192.168.0.xxx
Nov  8 13:15:45 nextcloudpi dnsmasq[11696]: config paradigma.remoteportal.de is 192.168.0.yyy

But the UDP packages do not appear. I'm complete stuck.

**Comment by Tobi7241** [comment]
one suggestion would be to use a small 4-port switch connecting SystaPi ENC28j60 port with SysaComfortII and pellet boiler to create a small own network segment. Then you can use the defauls network setup from that project ?
